src/array_api_stubs/_draft/searching_functions.py

```diff
@@ -87,6 +87,35 @@ def nonzero(x: array, /) -> Tuple[array, ...]:
     """
 
 
+def searchsorted(
+    x1: array,
+    x2: array,
+    /,
+    *,
+    side: Literal["left", "right"] = "left",
+    sorter: Optional[array] = None,
+) -> array:
+    """
+    Finds the indices into ``x1`` such that, if the corresponding elements in ``x2`` were inserted before the indices, the order of ``x1``, when sorted in ascending order, would be preserved.
+
+    Parameters
+    ----------
+    x1: array
+        input array. Must be a one-dimensional array. If ``sorter`` is ``None``, must be sorted in ascending order; otherwise, ``sorter`` must be an array of indices that sort ``x1`` in ascending order.
+    x2: array
+        array containing search values.
+    side: Literal['left', 'right']
+        if ``'left'``, then each returned index ``i`` must satisfy ``x1[i-1] < x2[m][n]...[j] <= x1[i]``, and, if, for an element ``x2[m][n]...[j]``, no index satisfies the index condition, then the returned index for that element must be ``0``. Otherwise, if ``'right'``, then each returned index ``i`` must satisfy ``x1[i-1] <= x2[m][n]...[j] < x1[i]``, and, if, for an element ``x2[m][n]...[j]``, no index satisfies the index condition, then the returned index for that element must be ``N``, where ``N`` is the number of elements in ``x1``. Default: ``'left'``.
+    sorter: Optional[array]
+        array of indices that sort ``x1`` in ascending order. The array must have the same shape as ``x1`` and have an integer data type. Default: ``None``.
+
+    Returns
+    -------
+    out: array
+        an array of indices with the same shape as ``x2``. The returned array must have the default array index data type.
+    """
+
```

Comment:
I am wondering if its worth it to add a note regarding the treatment of `NaN` values?

https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html
NumPy has two sentences about it. Unfortunately in CuPy we don't have it documented though...

Reply:
@emcastillo `NaN` sort order is discussed in the specification: https://data-apis.org/array-api/2022.12/API_specification/sorting_functions.html.

As in other functions, such as `sort` and `argsort`, the sort order of NaNs and signed zeros is implementation-dependent. In which case, I don't think we have anything normative to add here.

However, I will add a note indicating that order is likely to vary between implementations.

Reply:
Following-up on this, NumPy sorts `NaN` values to the end. However, the specification leaves the sort order of `NaN` values as unspecified. In which case, the following NumPy array would be a valid sort according to the specification:

```
In [1]: x = np.asarray([1.0, 2.0, np.nan, 3.0, 4.0, np.nan, 5.0])

In [2]: x
Out[2]: array([ 1.,  2., nan,  3.,  4., nan,  5.])
```

When invoking `np.searchsorted` on this array, NumPy always sorts search values to indices before `NaN` elements.

```
In [3]: np.searchsorted(x, np.asarray([0.5, 1.5, 2.5, 3.5, 4.5, 5.5]))
Out[3]: array([0, 1, 2, 4, 5, 7])

In [4]: np.searchsorted(x, np.asarray([0.5, 1.5, 2.5, 3.5, 4.5, 5.5]), side="left")
Out[4]: array([0, 1, 2, 4, 5, 7])

In [5]: np.searchsorted(x, np.asarray([0.5, 1.5, 2.5, 3.5, 4.5, 5.5]), side="right")
Out[5]: array([0, 1, 2, 4, 5, 7])
```

This behavior does violate the index condition; namely, that each returned index must satisfy `x1[i-1] < v <= x1[i]`, as `v < NaN` is always `False`.

I'll raise this concern in the upcoming workgroup meeting.

Reply:
Is there anything you still want to change here @kgryte?

Reply:
I added a note concerning real-valued floating-point sort order as relates to `NaNs` and signed zeros. Namely, as sort order is implementation-dependent, by implication, so too is `searchsorted` behavior. However, returned indices should be consistent with `sort` and `argsort`.